projects/packages/waf/src/class-waf-constants.php

```diff
@@ -61,9 +61,10 @@ public static function define_wpconfig_path() {
 	 */
 	public static function define_killswitch() {
 		if ( ! defined( 'DISABLE_JETPACK_WAF' ) ) {
-			$is_wpcom  = defined( 'IS_WPCOM' ) && IS_WPCOM;
-			$is_atomic = ( new Host() )->is_atomic_platform();
-			define( 'DISABLE_JETPACK_WAF', $is_wpcom || $is_atomic );
+			$is_wpcom        = defined( 'IS_WPCOM' ) && IS_WPCOM;
+			$is_atomic       = ( new Host() )->is_atomic_platform();
+			$is_atomic_on_jn = defined( 'IS_ATOMIC_JN' ) ?? IS_ATOMIC_JN;
+			define( 'DISABLE_JETPACK_WAF', $is_wpcom || ( $is_atomic && ! $is_atomic_on_jn ) );
```

Comment:
Aside: I wonder if we could use `Waf_Runner::is_supported_environment()` or an abstraction of it here so we don't have to duplicate these environment checks.